actions.js

```diff
@@ -0,0 +1,115 @@
+var store = require('./store.js');
+var types = require('./constants/action-types.js');
+
+function nextPage() {
+    var state = store.getState();
+    if(!state.duringPageFadeIn && state.completedPages[state.currentPage - 1]){
+        store.dispatch({
+            type: types['MOVE_PAGE'],
+            data: Math.min(state.duringPageFadeOut ? state.nextPage + 1: state.currentPage + 1, 7)
+        });
+    }
+}
+
+function prevPage() {
+    var state = store.getState();
+    if(!state.duringPageFadeIn) {
+        store.dispatch({
+            type: types['MOVE_PAGE'],
+            data: Math.max(state.duringPageFadeOut ? state.nextPage - 1 : state.currentPage - 1, 0)
+        });
+    }
+}
+
+function productTypeChange() {
+    store.dispatch({
+            type: "UPDATE_PRODUCT_TYPE",
+            data: $(this).val()
```

Comment:
you shouldn't use "dirty" functions as actions. pass value as argument:
```js
function productTypeChange(value) {
    store.dispatch({
        type: "UPDATE_PRODUCT_TYPE",
        data: value
    });
}

$(".foo").on("change", function (e) {
    productTypeChange($(this).val());
});
```
It will be more clear and testable.